validator/schema.go

```diff
@@ -199,6 +199,11 @@ func validateDefinition(schema *Schema, def *Definition) *gqlerror.Error {
 		if len(def.Fields) == 0 {
 			return gqlerror.ErrorPosf(def.Position, "%s must define one or more fields.", def.Kind)
 		}
+		for _, field := range def.Fields {
+			if typ, ok := schema.Types[field.Type.Name()]; ok && typ.Kind == InputObject {
```

Comment:
Maybe this should actually be a whitelist of accepted kinds?  I guess the error message is a little more obvious as it is though.